src/Qir/Execution/Tools/Executable/QirFullStateExecutable.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Quantum.Qir.Tools.Driver;
+using Microsoft.Quantum.Qir.Utility;
+
+namespace Microsoft.Quantum.Qir.Tools.Executable
+{
+    /// <summary>
+    /// Class to create and run QIR-based executables that use the full-state simulator.
+    /// </summary>
+    public class QirFullStateExecutable : QirExecutable
+    {
+        public QirFullStateExecutable(FileInfo executableFile, byte[] qirBytecode, ILogger logger, IQirDriverGenerator driverGenerator, IQirExecutableGenerator executableGenerator, IQuantumExecutableRunner runner)
```

Comment:
This class is going to be distributed as part of the QDK, and for the people that are just trying to use it a simpler constructor would be ideal. The simplest constructor would just take `executableFile` and `qirBytecode` as arguments:
```csharp
public QirFullStateExecutable(FileInfo executableFile, byte[] qirBytecode)
{
}
```
For testing purposes, you can create an `internal` constructor similar to this one that takes the additional arguments:
```csharp
internal QirFullStateExecutable(FileInfo executableFile, byte[] qirBytecode, ILogger logger, IQirDriverGenerator driverGenerator, IQirExecutableGenerator executableGenerator, IQuantumExecutableRunner runner)
{
}
```

Reply:
We use `InternalsVisibleTo` to make `internal` methods available to test projects.